specs/carousel/carousel.md

```diff
@@ -0,0 +1,258 @@
+    class="next-flipper"
+>
+    <svg class="flipper-svg">
+        <path d="example path..."></path>
+    </svg>
+</button>
+```
+
+**Default Play Toggle**
+```html
+<button
+    slot="play-toggle"
+    @click="${x => x.togglePlay()}"
+    class="previous-flipper"
+>
+    <svg class="play-svg">
+        <path d="example path..."></path>
+    </svg>
+</button>
+```
+
+**Default Tab List**
+```html
+<div role="tablist" class="tablist" aria-label="A carousel of items">
+    <div tabindex="0" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+</div>
+```
+
+**Implementation**
+```html
+<body>
+    <fast-carousel
+        // TODO: CHECK is this a way to apply the items?
+        item-0='<img src="...image1.src">'
+        item-1='<img src="...image2.src">'
+        item-2='<img src="...image3.src">'
+    >
+        <button
+            slot="next-flipper"
+            @click="${x => x.next()}" // TODO: How do I attach the carousel.next() function to a slotted item?
+            class="next-flipper"
+        >
+            <svg class="flipper-svg">
+                <path d="example path..."></path>
+            </svg>
+        </button>
+
+        <button
+            slot="previous-flipper"
+            @click="${x => x.next()}"
+            class="previous-flipper"
+        >
+            <svg class="flipper-svg">
+                <path d="example path..."></path>
+            </svg>
+        </button>
+
+        <button
+            slot="play-toggle"
+            @click="${x => x.togglePlay()}"
+            class="previous-flipper"
+        >
+            <svg class="play-svg">
+            <path d="example path..."></path>
+            </svg>
+        </button>
+
+        <div role="tablist" class="tablist" aria-label="A carousel of items">
+            <div tabindex="0" role="tab" class="tablist-tab"></div>
+            <div tabindex="-1" role="tab" class="tablist-tab"></div>
+            <div tabindex="-1" role="tab" class="tablist-tab"></div>
+            <div tabindex="-1" role="tab" class="tablist-tab"></div>
+            <div tabindex="-1" role="tab" class="tablist-tab"></div>
+        </div>
+    >
+    </fast-carousel>
+</body>
+```
+
+**Appearance:**
+| State | Image |
+| ----- | ----- |
+| default | ![](./images/carousel-light.png) |
+| dark mode | ![](./images/carousel-dark.png) |
+
+Parts:
+- carousel - the container for all the elements
+- carousel-item - the container for an item as content
+
+Animation:
+
+// TODO: ??? Same as Dialog or others?
+
+---
+
+## Implementation
+
+### States
+
+*Key component states, valid state transitions, and how interactions trigger a state transition.*
+
+- `isPaused` - if the carousel slide rotation is paused
+- `activeId` - the id of the active slide
```

Comment:
It might be useful to just store a reference to the active slide element itself.

Reply:
Added a reference prop